crates/context/config/src/client/env/proxy/query/proposal_approvers.rs

```diff
@@ -44,10 +48,44 @@ impl Method<Starknet> for ProposalApproversRequest {
     type Returns = Vec<ContextIdentity>;
 
     fn encode(self) -> eyre::Result<Vec<u8>> {
-        todo!()
+        // Get the full 32 bytes
+        let bytes = self.proposal_id.as_bytes();
+
+        // Split into high and low parts (16 bytes each)
+        let (high_bytes, low_bytes) = bytes.split_at(16);
+
+        // Create Felts with proper padding
+        let mut high = [0u8; 32];
+        let mut low = [0u8; 32];
+        high[16..].copy_from_slice(high_bytes); // Put in last 16 bytes
+        low[16..].copy_from_slice(low_bytes); // Put in last 16 bytes
+
+        let starknet_id = StarknetProposalId(FeltPair {
+            high: Felt::from_bytes_be(&high),
+            low: Felt::from_bytes_be(&low),
+        });
+
+        // Encode exactly as in mutate response
+        let mut encoded = Vec::new();
+        encoded.extend_from_slice(&starknet_id.0.high.to_bytes_be());
+        encoded.extend_from_slice(&starknet_id.0.low.to_bytes_be());
+        Ok(encoded)
```

Comment:
this could've been:

```suggestion
        let starknet_id = StarknetProposalId::from(self.proposal_id);

        let mut encoded = Vec::new();
        starknet_id.encode(&mut encoded);

        Ok(encoded)
```